### ISSUE TITLE
New message not showing without Refreshing tab or swithcing channels/DM

### ISSUE THREAD

**Issue by nilpatel42** [The-Commit-Company/Raven#777]
New message not showing without Refresh or need to change DM or channel to get new message and also counter of new message is not showing

**Comment by nikkothari22** [comment]
@nilpatel42 Looks like Socket is not working on your setup. 

Can you share your Frappe version, Raven version/branch, and if you've hosted on Frappe Cloud or self hosted? 

Can you also check if socket events are working? Just taking a screenshot of the browser console would be good enough.

**Comment by nilpatel42** [comment]
@nikkothari22 version of frappe and raven

ERPNext: v14.65.7 (version-14)

Frappe Framework: v14.68.2 (version-14)

Raven: v1.4.4 (develop)

**Comment by nikkothari22** [comment]
@nilpatel42 Can you check if socketIO is working? 

v14 is supported by Raven, and the develop branch is running fine on our setups (albeit v15).

**Comment by nilpatel42** [comment]
i don't know how to check it 

can you just give me step for it ?

thank you @nikkothari22

**Comment by nikkothari22** [comment]
@nilpatel42 

1. On (chrome), right click and select "Inspect".
2. Go to the "Console" tab (2nd tab) and see if there are any errors.

You can also check if the socket process is running on your self hosted server by running `sudo supervisorctl status all`

**Comment by nilpatel42** [comment]
@nikkothari22 

Via browser, its blank in console (while raven is open)

and when run `sudo supervisorctl status all `this command, 

Output :

```
● supervisor.service - Supervisor process control system for UNIX
     Loaded: loaded (/lib/systemd/system/supervisor.service; enabled; vendor preset: enabled)
     Active: active (running) since Fri 2024-02-23 12:04:26 IST; 4 weeks 0 days ago
       Docs: http://supervisord.org
   Main PID: 763 (supervisord)
      Tasks: 1 (limit: 9303)
     Memory: 3.8M
        CPU: 3min 54.911s
     CGroup: /system.slice/supervisor.service
             └─763 /usr/bin/python3 /usr/bin/supervisord -n -c /etc/supervisor/supervisord.conf

Notice: journal has been rotated since unit was started, output may be incomplete.
~
```

**Comment by nikkothari22** [comment]
> this command don't give any output sudo supervisorctl status all
> 


Did you run this inside the bench directory on your server? 

I'll install the develop branch on v14 just to be sure that it's working and let you know.

**Comment by nilpatel42** [comment]
Yes i run this command inside my frappe bench directory  

`abc@abc:~/frappe-bench$`

**Comment by nikkothari22** [comment]
Oh so is this running on developer mode on your local machine, or in production on a server?

Supervisor should be running on a production setup 🤔

**Comment by nilpatel42** [comment]
I have production server

**Comment by nikkothari22** [comment]
@nilpatel42 - can you join this community here; https://raven-community.frappe.cloud . Would be better to debug it there.

**Comment by nikkothari22** [comment]
Issue was with socket events not being fired for the administrator account on v14 for some reason. This should be fixed in the next release via #745
